src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/DuplicateTypeNameAnalyzer.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Azure.ClientSdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DuplicateTypeNameAnalyzer : SymbolAnalyzerBase
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Descriptors.AZC0034);
+        public override SymbolKind[] SymbolKinds { get; } = { SymbolKind.NamedType };
+
+        // Common .NET platform type names that Azure SDK types should avoid conflicting with
+        private static readonly HashSet<string> PlatformTypeNames = new HashSet<string>
+        {
+            // Common System types
+            "Object", "String", "Boolean", "Char", "Byte", "SByte", "Int16", "UInt16", "Int32", "UInt32", "Int64", "UInt64",
```

Comment:
yeah, as I said above. I don't think all the names should be hardcoded here in the source.